pallets/loans/src/rate_model.rs

```diff
@@ -165,19 +185,57 @@ impl CurveModel {
     pub fn new_model(base_rate: Rate) -> CurveModel {
         Self { base_rate }
     }
+}
 
-    /// Check the curve model for sanity
-    pub fn check_model(&self) -> bool {
-        true
+impl InterestRateModelLike for CurveModel {
+    fn check_model(&self) -> bool {
+        false
     }
 
-    /// Calculates the borrow interest rate of curve model
-    pub fn get_borrow_rate(&self, _utilization: Ratio) -> Option<Rate> {
-        // TODO: Need to implement the curve model
+    fn get_borrow_rate(&self, _: Ratio) -> Option<Rate> {
         None
     }
 }
 
+#[derive(Clone, Copy, Decode, Default, Encode, Eq, PartialEq, RuntimeDebug)]
+pub struct NewCurveNameModel {
+    borrow_rate: Rate,
+    reserve_ratio: Ratio,
+    staking_apy: Ratio,
```

Comment:
reserve_ratio should use the ReserveFactor storage.

Reply:
Renamed to `reserve_ratio` to remember that its value should come from the `ReserveFactor` storage